### PR TITLE
Hook tf.tensor

### PR DESCRIPTION
**Pull request by bendecoste**

### CODE REVIEW

File: syft_tensorflow/hook/hook.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 from syft_tensorflow.attributes import TensorFlowAttributes
 from syft_tensorflow.tensor import TensorFlowTensor
+from syft_tensorflow.attributes import TensorFlowAttributes
```
Comment (yanndupis) [inline syft_tensorflow/hook/hook.py]: We are import two times `TensorFlowAttributes`

File: syft_tensorflow/attributes/attributes.py [hunk 2]
```diff
@@ -32,11 +32,12 @@ class TensorFlowAttributes(FrameworkAttributes):
...
     Tensor = TensorFlowTensor
 
-    def __init__(self, tensorflow: ModuleType, hook: "TensorFlowHook"):
+    def __init__(self, tensorflow: ModuleType, hook: ModuleType):
+        super().__init__(tensorflow, hook)
         # Stash the hook here for global access elsewhere
         self.hook = hook
 
-        # SECTION: List all functions in tf module that we want to overload
+        self.allowed_commands = set()
```
Comment (yanndupis) [inline syft_tensorflow/attributes/attributes.py]: Do we need this attribute def here? later in the script we do `self.allowed_commands = self._tensorflow_modules_functions`

File: syft_tensorflow/attributes/attributes.py [hunk 3]
```diff
@@ -32,12 +32,11 @@ class TensorFlowAttributes(FrameworkAttributes):
     ALIAS = "tensorflow"
     Tensor = TensorFlowTensor
 
-    def __init__(self, tensorflow: ModuleType, hook: "TensorFlowHook"):
+    def __init__(self, tensorflow: ModuleType, hook: ModuleType):
```
Comment (jvmancuso) [inline syft_tensorflow/attributes/attributes.py]: This should be changed back to "TensorFlowHook" for the `hook` argument type hint

File: syft_tensorflow/attributes/attributes.py [hunk 4]
```diff
@@ -107,9 +106,16 @@ def __init__(self, tensorflow: ModuleType, hook: "TensorFlowHook"):
 
         self.command_guard = self._command_guard
 
-        # Dict {method_name: <is_inplace:bool>
+        self.exclude = [
+            '__getattr__',
```
Comment (jvmancuso) [inline syft_tensorflow/attributes/attributes.py]: Why are we excluding `__getattr__`? Also the attributes below should not be here unless totally necessary.

File: syft_tensorflow/hook/hook.py [hunk 5]
```diff
@@ -21,13 +21,15 @@ def __init__(
         local_worker: BaseWorker = None,
         is_client: bool = True
     ):
-        self.tensorflow = tensorflow
-        self.tensorflow.hook = self
+
+        self.tensorflow = tf
```
Comment (jvmancuso) [inline syft_tensorflow/hook/hook.py]: We want to assign the module that's passed in via the tensorflow arg and not the tf module that's imported above, so this needs to be fixed.

**Pull request merged**